apps/app/src/components/Timer.tsx

```diff
@@ -0,0 +1,52 @@
+import { Timer } from "@/types/Timer";
+import { leftPad } from "@/utilities/global";
+import { useEffect, useState } from "react";
+import { Text, View } from "react-native";
+
+interface Props {
+  start: Date;
+  children: string;
+}
+
+const initialTimer: Timer = {
+  hours: "00",
+  minutes: "00",
+  seconds: "00",
+};
+
+const countUp = (start: Date): Timer => {
+  const now = new Date();
+  const distance = now.getTime() - start.getTime();
+
+  const time: Timer = {
+    hours: leftPad(Math.floor(distance / (1000 * 60 * 60))),
+    minutes: leftPad(Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60))),
+    seconds: leftPad(Math.floor((distance % (1000 * 60)) / 1000)),
+  };
```

Comment:
Here you can extract a `Timer` class as well (instead of a type), because it contains logic.
Maybe with the earlier example you can figure this one out yourself (let me know if you need help).

You'd want to end up with a class that will allow you to do something like this:

```typescript
const timer = new Timer(Date.now())

// ... wait some time

// These have to be functions because when you call them they have to calculate the time based on `Date.now()`.
console.log(timer.hours())
console.log(timer.minutes())
console.log(timer.seconds())

console.log(timer.seconds()) // This could give a different result than the one above, because it's calculated again.. if this happens to be executed on the next second. It's an edge-case, but it will happen at some point.

// Maybe better would be:
const time = timer.diff(Date.now()) // Calculate the difference between input of this function and the start of the timer
console.log(time.hours) // These values don't change, because they were calculated once and stored in the time variable
console.log(time.minutes)
console.log(time.seconds)

```